src/main/lombok/eu/coinform/gateway/controller/CheckController.java

```diff
@@ -171,6 +202,21 @@ public void corsHeadersEvaluate(HttpServletResponse response) {
         response.addHeader("Access-Control-Max-Age", "3600");
     }
 
+    @RequestMapping(value = "/twitter/evaluate/label", method = RequestMethod.POST)
+    public ResponseEntity<?> evaluateLabel(@Valid @RequestBody TweetLabelEvaluationForm tweetLabelEvaluationForm) {
+        SecurityContext context = SecurityContextHolder.getContext();
+        Authentication authentication = context.getAuthentication();
+        Long userId = (Long) authentication.getPrincipal();
+        Optional<User> oUser = userDbManager.getUserById(userId);
+
+        if(oUser.isEmpty()){
+            log.debug("No user: {}, {}", userId, authentication.getPrincipal());
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ErrorResponse.NOSUCHUSER);
+        }
+        eventPublisher.publishEvent(new UserLabelReviewEvent(new LabelEvaluationImplementation(tweetLabelEvaluationForm, oUser.get().getUuid())));
+        return ResponseEntity.ok(SuccesfullResponse.EVALUATELABEL);
+    }
+
```

Comment:
Both evaluateLabel, evaluateTweet are executed sequentially, shouldn't the events be called async?

Reply:
fixed. now async